boards/pic32-clicker/include/periph_conf.h

```diff
@@ -59,6 +62,48 @@ extern "C" {
 #define DEBUG_UART_BAUD     (9600)
 /** @} */
 
+
+/**
+ * @name    SPI device configuration
+ *
+ * @{
+ */
+
+static const spi_conf_t spi_config[] = {
+    {}, /* No SPI0 on PIC32 */
```

Comment:
what does that mean? why not start the array at zero?

Reply:
By not starting the array at zero,```spi_bus_t``` matches the index of the SPI module so you can write ```spi_init(1)``` to initialise SPI module 1. However, I noticed that ```periph_init``` initialises variables from ```SPI_DEV(0)``` to ```SPI_DEV(SPI_NUMOF-1)``` so this convention would not work with this function as currently ```SPI_DEV(0)``` does not exist.

What is the convention in RIOT about this ? Are other peripherals indexed from 0 ?
Note that if we change to a 0-based index, we would also need to modify uart peripheral in mips_pic32_common.

Reply:
Annoyingly Microchip number their peripherals from 1 not 0, so although this looks a very odd thing to do, if you don't do this it gets even more confusing when you're accessing registers (defined by the vendor supplied register files for the device) named DEVICE1 for logical device 0.

I would prefer to keep the device numbering as the UART driver ie 1->X with device 0 empty, adding assertions in the driver code to check device 0 never gets used.